### PR TITLE
Delayed neutron precursor decay rate tally

### PR DESCRIPTION
**Pull request by samuelshaner**
This PR adds a tally for the delayed neutron precursor decay rate (also called the decay constant). This tally is important for kinetics analysis with any cross section library that does not have the same decay rates array for each nuclide (e.g. nndc). Some important notes about the implementation of this tally:

* The tally only supports an analog estimator since the decay rate for a specific nuclide is needed in the tally. I guess one could loop over all the nuclides in a material for a track length tally, but that seems onerous.
* The decay rates in the nndc ace files seem to be in units of *shakes* and not *seconds*. This was deduced by looking at the tallied decay rates and not any documentation. I searched around for any reference of the units on the decay rate, but could not find a definitive answer anywhere. If anyone has any info on whether there is a difference in units in different libraries, please do share!
* I'm not positive on the correct procedure to tally the decay rates. I chose to tally the delayed-nu-fission weighted decay rate as delayed fissions produce delayed neutron precursors. This would effectively weight the decay rate by the delayed neutron precursor production rate. However, another option would be to tally the mean lifetime of a delayed neutron precursor times the delayed-nu-fission rate and then invert this to get the decay rate. 
* I implemented a MDGXS for the decay rate and added an example of computing decay rates to `mdgxs-part-i.ipynb`.
* The documentation and test suite have been updated. 
* This PR builds on PR #711 

openmc_decay_rate

### CODE REVIEW

File: openmc/mgxs/mdgxs.py [hunk 1]
```diff
@@ -1574,3 +1577,155 @@ def xs_tally(self):
...
+    :class:`openmc.StatePoint` instance. The derived multi-group cross section
+    can then be obtained from the :attr:`DecayRate.xs_tally` property.
+
+    For a spatial domain :math:`V`, energy group :math:`[E_g,E_{g-1}]`, and
+    delayed group :math:`d`, the decay rate is calculated as:
+
+    .. math::
+
+       \langle \lambda_d \nu^d \sigma_f \phi \rangle &= \int_{r \in V} dr
+       \int_{4\pi} d\Omega' \int_0^\infty dE' \int_0^\infty dE \; \chi(E) \nu^d
```
Comment (paulromano) [inline openmc/mgxs/mdgxs.py]: I don't see the decay rate anywhere on the RHS

File: openmc/mgxs/mdgxs.py [hunk 2]
```diff
@@ -1574,3 +1577,155 @@ def xs_tally(self):
...
+       \sigma_f (r, E') \psi(r, E', \Omega') \\
+       \langle \nu \sigma_f \phi \rangle &= \int_{r \in V} dr \int_{4\pi}
```
Comment (paulromano) [inline openmc/mgxs/mdgxs.py]: Should this be `\langle \nu^d ...`?

File: openmc/mgxs/mdgxs.py (hunk 1 above)
Comment (paulromano) [inline openmc/mgxs/mdgxs.py]: Also, why does chi appear in these equations if it's just immediately integrated out?

File: src/tally.F90 [hunk 3]
```diff
@@ -670,6 +672,143 @@ subroutine score_general_ce(p, t, start_index, filter_index, i_nuclide, &
...
+                       % nu(E, EMISSION_DELAYED, d)
+
+                  associate (rxn => nuclides(p % event_nuclide) % &
+                       reactions(nuclides(p % event_nuclide) % index_fission(1)))
+
+                    ! Compute the score
+                    score = p % absorb_wgt * yield * &
+                         micro_xs(p % event_nuclide) % fission &
+                         / micro_xs(p % event_nuclide) % absorption &
+                         * rxn % products(1 + d) % decay_rate * 1.e8
```
Comment (paulromano) [inline src/tally.F90]: Use `1.e8_8` to make sure it's a double

**Comment by paulromano** [comment]
Looks like there are some conflicts on the MGXS library tests (just need to update the inputs). Otherwise I think this looks fine.

One thought -- why don't we just have the Python API convert the decay rates from ACE files to seconds? No need to adopt the weird shake units. And we might as well mess around with the HDF5 format now before it starts to stabilize.

**Comment by paulromano** [comment]
You also need to add the new class in `docs/source/pythonapi/index.rst`

File: openmc/mgxs/mdgxs.py (hunk 2 above)
Comment (samuelshaner) [inline openmc/mgxs/mdgxs.py]: yep

**Comment by samuelshaner** [comment]
@paulromano - I just merged with upstream/develop and now all but 5 of the tests are failing. The merge conflicts I changed were with the MGXS library tests. I looked through all the files and don't see any reason why all the tests would now fail. Do you (or anyone else) have any thoughts?

**Comment by nelsonag** [comment]
It's probably a cache clearing issue since develop has new h5 libraries. So
just clear Travis' cache

**Comment by nelsonag** [comment]
FYI so you're not confused by any Travis-magic, I was clearing on the
travis site anyways so I just cleared the cache for your PR.  It was
passing afterwards

**Comment by samuelshaner** [comment]
@nelsonag - thanks for clearing the cache! Glad it was something simple.

@paulromano - you asked why don't we just have the Python API convert the decay rates from ACE files to seconds. I modified the `openmc/data/product.py` to convert the decay rate from units of shakes to seconds and then removed the conversion in `tally.F90`. Do you know if all cross section libraries have the decay rates in units of shakes or should I have an option to allow the user to set the units of decay rate in their cross section library? In making this addition, I believe I need to regenerate the hdf5 cross section files. Is that true? If so, how do I go about updating them and having that update reflected in the cross sections used by the test suite run on travis?

**Comment by paulromano** [comment]
All ACE files have decay rates in inverse shakes and all ENDF files have decay rates in inverse seconds. Right now I would leave out any option to change units since we are units-oblivious right now (units are never stored anywhere -- this probably ought to change in the future). As for regenerating the HDF5 data, if you push the change to `product.py`, I can go ahead and regenerate the NNDC library, upload it to box, and send you the URL which you can use to update `.travis.yml`.

**Comment by paulromano** [comment]
@samuelshaner Can you make the following hotfix? This will fix your Python 2 MCNP data conversion bug.
```diff
diff --git a/openmc/data/neutron.py b/openmc/data/neutron.py
index 456ccee..2431ba7 100644
--- a/openmc/data/neutron.py
+++ b/openmc/data/neutron.py
@@ -478,7 +478,7 @@ class IncidentNeutron(EqualityMixin):
                 if len(rxs) > 0:
                     data.summed_reactions[mt_sum] = rx = Reaction(mt_sum)
                     for T in data.temperatures:
-                        rx.xs[T] = Sum([rx.xs[T] for rx in rxs])
+                        rx.xs[T] = Sum([rx_i.xs[T] for rx_i in rxs])
 
         # Read unresolved resonance probability tables
         if 'urr' in group:
```

**Comment by samuelshaner** [comment]
@paulromano - just added your hotfix.

File: openmc/data/product.py [hunk 4]
```diff
@@ -98,7 +98,9 @@ def applicability(self, applicability):
     def decay_rate(self, decay_rate):
         cv.check_type('product decay rate', decay_rate, Real)
         cv.check_greater_than('product decay rate', decay_rate, 0.0, True)
-        self._decay_rate = decay_rate
+
+        # Convert the decay rate from units of inverse shakes to inverse seconds
+        self._decay_rate = decay_rate * 1.e8
```
Comment (paulromano) [inline openmc/data/product.py]: I would actually leave this as it was before and instead put the unit conversion on line 110 of reaction.py.

Comment (samuelshaner) [inline openmc/data/product.py]: ok, will do.

**Comment by paulromano** [comment]
Thanks @samuelshaner!

**Comment by samuelshaner** [comment]
Thanks for all the help @paulromano!

**Pull request merged**